tooltool.py

```diff
@@ -997,28 +946,26 @@ def main():
     parser.add_option('--authentication-file',
                       help='Use http authentication to download a file.', dest='auth_file')
 
-    (options_obj, args) = parser.parse_args()
+    (options_obj, args) = parser.parse_args(argv[1:])
     # Dictionaries are easier to work with
     options = vars(options_obj)
-    # Use some of the option parser to figure out application
-    # log level
-    if options.get('verbose'):
-        log.setLevel(logging.DEBUG)
-    elif options.get('quiet'):
-        log.setLevel(logging.ERROR)
-    else:
-        log.setLevel(logging.INFO)
-    log.addHandler(ch)
 
-    if 'manifest' not in options:
-        parser.error("no manifest file specified")
+    log.setLevel(options['loglevel'])
+
+    # Set up logging, for now just to the console
+    if not _skip_logging:  # pragma: no cover
+        ch = logging.StreamHandler()
+        cf = logging.Formatter("%(levelname)s - %(message)s")
+        ch.setFormatter(cf)
+        log.addHandler(ch)
+
+    if options['algorithm'] != 'sha512':
+        parser.error('only --algorithm sha512 is supported')
```

Comment:
I was going to say "Let's get rid of --algorithm", but we should probably keep it to be backward compatible.
I'd add a deprecation warning somewhere.